File: MyLanguageWeb/src/main/java/ru/herooo/mylanguageweb/dto/entity/workoutitem/other/AnswerInfoResponseDTO.java
package ru.herooo.mylanguageweb.dto.entity.workoutitem.other;

import com.fasterxml.jackson.annotation.JsonProperty;
import ru.herooo.mylanguageweb.dto.entity.workoutitem.response.WorkoutItemResponseDTO;

import java.util.List;

public class AnswerInfoResponseDTO {

    @JsonProperty("workout_item")
    private WorkoutItemResponseDTO workoutItem;

    @JsonProperty("message")
    private String message;

    @JsonProperty("is_correct")
    private boolean isCorrect;

    @JsonProperty("possible_answers")
    private List<String> possibleAnswers;


    public WorkoutItemResponseDTO getWorkoutItem() {
        return workoutItem;
    }

    public void setWorkoutItem(WorkoutItemResponseDTO workoutItem) {
        this.workoutItem = workoutItem;
    }

    public boolean getIsCorrect() {
        return isCorrect;
    }

    public void setIsCorrect(boolean correct) {
        isCorrect = correct;
    }

    public List<String> getPossibleAnswers() {
        return possibleAnswers;
    }

    public void setPossibleAnswers(List<String> possibleAnswers) {
        this.possibleAnswers = possibleAnswers;
    }

    public String getMessage() {
        return message;
    }

    public void setMessage(String message) {
        this.message = message;
    }
}

File: MyLanguageWeb/src/main/resources/static/js/classes/dto/entity/workout_item/other/answer_info_response_dto.js
import {
    WorkoutItemResponseDTO
} from "../response/workout_item_response_dto.js";

export class AnswerInfoResponseDTO {
    #message;
    #isCorrect;
    #possibleAnswers;
    #workoutItem;

    constructor(json) {
        if (json) {
            this.#message = json["message"];
            this.#isCorrect = json["is_correct"];
            this.#possibleAnswers = json["possible_answers"];

            let workoutItem = json["workout_item"];
            if (workoutItem) {
                this.#workoutItem = new WorkoutItemResponseDTO(workoutItem);
            }
        }
    }

    getMessage() {
        return this.#message;
    }

    getIsCorrect() {
        return this.#isCorrect;
    }

    getPossibleAnswers() {
        return this.#possibleAnswers;
    }

    getWorkoutItem() {
        return this.#workoutItem;
    }

    getPossibleAnswersStr() {
        let str;
        let possibleAnswers = this.#possibleAnswers;
        if (possibleAnswers) {
            str = possibleAnswers.join(", ");
        }

        return str;
    }
}